apps/relayer/src/transaction/transaction.service.spec.ts

```diff
@@ -188,11 +194,29 @@ describe('TransactionService', () => {
         }))
 
         expect(watchTransaction).toHaveBeenCalledWith(tx.hash, result)
-        // The receipt wait isn't waited on, it happens in the next tick
+
+        // Ensure the checkTransactions method is called
+        jest.advanceTimersToNextTimer(1)
+        await txPromise
+        expect(checkTransactions).toHaveBeenCalled()
+
+        // Shouldn't remove it from the unwatch list until it's finalized
+        await setTimeout(() => {
+          expect(unwatchTransaction).not.toHaveBeenCalledWith(tx.hash)
+        })
+
+        // Simulate being included in a block and ensure it's unwatched
+        const completedTx = txFixture()
+        completedTx.hash = tx.hash
+        completedTx.blockHash = "notNull"
+        const completedTxPromise = Promise.resolve(completedTx)
+        getTransaction.mockReturnValue(completedTxPromise)
+        jest.advanceTimersToNextTimer(2)
+        await completedTxPromise
```

Comment:
I noticed that the async calls within `checkTransactions` wasn't being executed until I called await explicitly here

Reply:
This describes it pretty well:
https://github.com/facebook/jest/issues/9181#issuecomment-567384171